tests/common/plugins/conditional_mark/tests_mark_conditions.yaml

```diff
@@ -1401,163 +1401,181 @@ qos/test_qos_sai.py::TestQosSai:
 
 qos/test_qos_sai.py::TestQosSai::testIPIPQosSaiDscpToPgMapping:
   skip:
-    reason: "For DSCP to PG mapping on IPinIP traffic , mellanox device has different behavior to community. For mellanox device, testQosSaiDscpToPgMapping can cover the scenarios / M0/MX topo does not support qos"
+    reason: "For DSCP to PG mapping on IPinIP traffic , mellanox device has different behavior to community. For mellanox device, testQosSaiDscpToPgMapping can cover the scenarios / Unsupported testbed type."
     conditions_logical_operator: or
     conditions:
       - "asic_type in ['mellanox']"
       - https://github.com/sonic-net/sonic-mgmt/issues/12906
       - "topo_type in ['m0', 'mx']"
+      - "topo_name not in ['t0', 't0-64', 't0-116', 't0-35', 't0-56', 't0-standalone-32', 't0-standalone-64', 't0-standalone-128', 't0-standalone-256', 'dualtor-56', 'dualtor-120', 'dualtor', 't0-80', 't0-backend', 't1-lag', 't1-64-lag', 't1-56-lag', 't1-backend', 't2', 't2_2lc_36p-masic', 't2_2lc_min_ports-masic'] and asic_type not in ['mellanox']"
```

Comment:
It looks tedious to enumerate all topology names here. And it could be easily broken if a new topology is introduced.
Can it be simplified by checking `topo_type`?

Reply:
This condition is a copy from the entry `qos/test_qos_sai.py::TestQosSai:`. I checked the topo_name list here, for example, for t0, it doesn't cover all topo_name of topo_type t0. If we use topo_type in ['t0'] to skip this case, I think we will skip more cases by mistake?

Reply:
`topo_type` will be `t0` for dualtor, if we use `topo_type`, it cannot differentiate between `t0` and `dualtor`

Reply:
We can take advantage of "AND" and "OR" to handle such special t0 type cases.